### PR TITLE
Feature/results page level 2 changes

### PR DESCRIPTION
**Pull request by DanielClarkeEducation**
# Description

Changes to the logic of the Qualification Details page.

There is now an edge case where If a qualification:

 - is not full and relevant (be it not relevant from Contentful, or not relevant calculated from question answers)
 - was started between September 2014 and August 2019
 - is above level 2

they might be able to work in a level 2 setting, but will need to enquire with the DfE.

The logic to present this information has been added with a new "Further action required" tag along with another set of requirement text.

## Ticket number (if applicable)

https://dfedigital.atlassian.net.mcas.ms/browse/EYQB-725

# How Has This Been Tested?

- Tested locally with live Contentful

- New unit tests for the logic and an E2E test to check the page displays the correct text.

- Existing test suite passing

# Screenshots

Please include screenshots if applicable.

# Checklist:

- [x] My code follows the standards used within this project
- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests (Unit, E2E) that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Dfe.EarlyYearsQualification.Web/Models/Content/RatioRequirementModel.cs [hunk 1]
```diff
@@ -2,18 +2,18 @@ namespace Dfe.EarlyYearsQualification.Web.Models.Content;
...
+    public QualificationApprovalStatus ApprovedForUnqualified { get; set; }
 
-    public bool IsNotFullAndRelevant
-    {
-        get { return ApprovedForUnqualified && !ApprovedForLevel2 & !ApprovedForLevel3 && !ApprovedForLevel6; }
-    }
+    public bool IsNotFullAndRelevant =>
+        ApprovedForLevel2 != QualificationApprovalStatus.Approved
+        && ApprovedForLevel3 != QualificationApprovalStatus.Approved
+        && ApprovedForLevel6 != QualificationApprovalStatus.Approved;
```
Comment (sam-c-dfe) [inline src/Dfe.EarlyYearsQualification.Web/Models/Content/RatioRequirementModel.cs:16]: If the value is now an enum should we also specify what the acceptable value is here for ApprovedForUnqualified?

File: src/Dfe.EarlyYearsQualification.Web/Services/UserJourneyCookieService/UserJourneyCookieService.cs [hunk 2]
```diff
@@ -210,6 +210,20 @@ public bool WasStartedOnOrAfterSeptember2014()
         return date >= new DateOnly(2014, 9, 1);
     }
 
+    public bool WasStartedBetweenSept2014AndAug2019()
```
Comment (sam-c-dfe) [inline src/Dfe.EarlyYearsQualification.Web/Services/UserJourneyCookieService/UserJourneyCookieService.cs:213]: Can you add some unit tests for this please? :-)

File: src/Dfe.EarlyYearsQualification.Web/Models/Content/RatioRequirementModel.cs (hunk 1 above)
Comment (DanielClarkeEducation) [reply src/Dfe.EarlyYearsQualification.Web/Models/Content/RatioRequirementModel.cs:16]: With every qualification being approved at unqualified, it seems a bit redundant to check it. I can add it is as an Approved check to be more verbose?

Comment (sam-c-dfe) [reply src/Dfe.EarlyYearsQualification.Web/Models/Content/RatioRequirementModel.cs:16]: Nah thinking about it more, leave it as is :-)

**Review by sam-c-dfe — APPROVED** [review]

**Review by RobertGHippo — APPROVED** [review]
A couple of minor comments, none of which I think should block this 🙂

File: src/Dfe.EarlyYearsQualification.Web/Models/Content/RatioRequirementModel.cs [hunk 3]
```diff
@@ -2,18 +2,18 @@ namespace Dfe.EarlyYearsQualification.Web.Models.Content;
...
+    public QualificationApprovalStatus ApprovedForLevel6 { get; set; }
 
-    public bool ApprovedForUnqualified { get; set; }
+    public QualificationApprovalStatus ApprovedForUnqualified { get; set; }
 
-    public bool IsNotFullAndRelevant
-    {
-        get { return ApprovedForUnqualified && !ApprovedForLevel2 & !ApprovedForLevel3 && !ApprovedForLevel6; }
-    }
+    public bool IsNotFullAndRelevant =>
```
[the remark this answers is not in the document]
Comment (DanielClarkeEducation) [reply src/Dfe.EarlyYearsQualification.Web/Models/Content/RatioRequirementModel.cs:13]: This one is Rider overwriting, I can change it back no problem

File: src/Dfe.EarlyYearsQualification.Mock/Content/MockQualificationsRepository.cs [hunk 4]
```diff
@@ -17,7 +17,11 @@ public class MockQualificationsRepository : IQualificationsRepository
                    "eyq-108" => await Task.FromResult(CreateQtsQualification("EYQ-108", "BTEC",
                                                                              AwardingOrganisations.Various, 6)),
                    "eyq-115" => await Task.FromResult(CreateQualification("EYQ-115", "NCFE",
-                                                                             AwardingOrganisations.Various, 3, false)),
+                                                                          AwardingOrganisations.Various, 3, false)),
+                   "eyq-114" => await Task.FromResult(CreateLevel2FurtherActionRequiredQualification("EYQ-114", "Level 2 Further Action Qualification", AwardingOrganisations.Ncfe, 3)),
+                   
+                   "eyq-241" => await Task.FromResult(CreateQualification("EYQ-241", "BTEC",
```
[the remark this answers is not in the document]
Comment (DanielClarkeEducation) [reply src/Dfe.EarlyYearsQualification.Mock/Content/MockQualificationsRepository.cs:23]: It is used in the E2E tests to cover existing tests, not hitting the new level 2 functionality

**Review by RobertGHippo — APPROVED** [review]

**Pull request merged**